pqarrow/arrowutils/sort.go

```diff
@@ -197,15 +187,47 @@ func newMultiColSorter(
 					},
 				}
 			default:
+				ms.Release()
```

Comment:
We could do named returns (which I don't particularly love) but then just have the defer func check if this function is about to return an error and then release ms if it is.